### PR TITLE
Add Doxygen config

### PR DESCRIPTION
**Pull request by oddstr13**
**Changes**
Add Doxygen config file

See [https://thoas.feralhosting.com/oddstr13/jellyfin/doxygen/](https://thoas.feralhosting.com/oddstr13/jellyfin/doxygen/class_media_browser_1_1_controller_1_1_entities_1_1_base_item.html) for example output.

### CODE REVIEW

**Comment by oddstr13** [comment]
Slight differences from the previous version, due to me initially generating the config with a older version of Doxygen. Should be functionally equal however.
I used the `Doxygen GUI frontend` to customize the config, after generating the template using the commandline `doxygen -g`.

File: Doxyfile [hunk 1]
```diff
@@ -0,0 +1,2565 @@
...
+# This tag requires that the tag FILTER_SOURCE_FILES is set to YES.
+
+FILTER_SOURCE_PATTERNS =
+
+# If the USE_MDFILE_AS_MAINPAGE tag refers to the name of a markdown file that
+# is part of the input, its contents will be placed on the main page
+# (index.html). This can be useful if you have a project on for instance GitHub
+# and want to reuse the introduction page also for the doxygen output.
+
+USE_MDFILE_AS_MAINPAGE =
```
Comment (LogicalPhallacy) [inline Doxyfile]: Might be worth specifying the GitHub readme.md here.

File: Doxyfile [hunk 2]
```diff
@@ -0,0 +1,2565 @@
...
+# The HTML_COLORSTYLE_HUE tag controls the color of the HTML output. Doxygen
+# will adjust the colors in the style sheet and background images according to
+# this color. Hue is specified as an angle on a colorwheel, see
+# https://en.wikipedia.org/wiki/Hue for more information. For instance the value
+# 0 represents red, 60 is yellow, 120 is green, 180 is cyan, 240 is blue, 300
+# purple, and 360 is red again.
+# Minimum value: 0, maximum value: 359, default value: 220.
+# This tag requires that the tag GENERATE_HTML is set to YES.
+
+HTML_COLORSTYLE_HUE    = 220
```
Comment (LogicalPhallacy) [inline Doxyfile]:
```suggestion
HTML_COLORSTYLE_HUE    = 195
```

This seems like a good place for Jellyfin Blue

File: Doxyfile [hunk 3]
```diff
@@ -0,0 +1,2565 @@
...
+# title of most generated pages and in a few other places.
+# The default value is: My Project.
+
+PROJECT_NAME           = Jellyfin
+
+# The PROJECT_NUMBER tag can be used to enter a project or revision number. This
+# could be handy for archiving the generated documentation or if some version
+# control system is used.
+
+PROJECT_NUMBER         =
```
[the remark this answers is not in the document]
Comment (oddstr13) [reply Doxyfile]: Token or not, should be easy enough to set in CI
```bash
$ VER=123
$ sed  -i "s/PROJECT_NUMBER *=.*/PROJECT_NUMBER = ${VER}/" Doxyfile
$ grep PROJECT_NUMBER Doxyfile
# The PROJECT_NUMBER tag can be used to enter a project or revision number. This
PROJECT_NUMBER = 123
```
If a token of sorts is desired, I'd need to know in what format it should be.

File: Doxyfile [hunk 4]
```diff
@@ -0,0 +1,2565 @@
...
+
+OPTIMIZE_OUTPUT_FOR_C  = NO
+
+# Set the OPTIMIZE_OUTPUT_JAVA tag to YES if your project consists of Java or
+# Python sources only. Doxygen will then generate output that is more tailored
+# for that language. For instance, namespaces will be presented as packages,
+# qualified scopes will look different, etc.
+# The default value is: NO.
+
+OPTIMIZE_OUTPUT_JAVA   = YES
```
[the remark this answers is not in the document]
Comment (oddstr13) [reply Doxyfile]: Yes, GUI says it's for Java and C#

File: Doxyfile (hunk 1 above)
Comment (oddstr13) [reply Doxyfile]: Was already rendered, but not set as main page.
As a side note, Doxygen is *not* happy with the html in the readme. (especially the status icons)

File: Doxyfile (hunk 2 above)
Comment (oddstr13) [reply Doxyfile]: Is this the color you wanted?
screenshot
New color to the right.

File: Doxyfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (oddstr13) [reply Doxyfile]: I don't know how doxygen handles this parameter.
Presumably it is just a arbitrary string presented in the output.
I'm re-running doxygen with `10.2.2` in that field right now.

[the remark this answers is not in the document]
Comment (oddstr13) [reply Doxyfile]: screenshot

[the remark this answers is not in the document]
Comment (oddstr13) [reply Doxyfile]: Do note that quotes are needed for strings with space, as far as I understand it. `"foo bar"`

[the remark this answers is not in the document]
Comment (joshuaboniface) [reply Doxyfile]: Looks good to me, should be trivial to update this during version bump.

**Pull request merged**